src/agent/solana/exporter.rs

```diff
@@ -285,12 +302,63 @@ impl Exporter {
             })
             .collect::<Vec<_>>();
 
-        if fresh_updates.is_empty() {
+        // Update permissioned accounts of this publisher from the
+        // oracle. The loop ensures that we clear the channel and use
+        // only the final, latest message; try_recv() is non-blocking.
+        loop {
+            match self.perm_price_rx.try_recv() {
+                Ok(perm_price_accounts) => {
+                    self.perm_price_accounts = perm_price_accounts;
+                    trace!(
+                        self.logger,
+                        "Exporter: read permissioned price accounts from channel";
+                        "new_value" => format!("{:?}", self.perm_price_accounts),
+                    );
+                }
+                // Expected failures when channel is empty
+                Err(TryRecvError::Empty) => {
+                    debug!(
+                        self.logger,
+                        "Exporter: No more permissioned price accounts in channel, using cached value";
+                        "cached_value" => format!("{:?}", self.perm_price_accounts),
+                    );
+                    break;
+                }
+                // Unexpected failures (channel closed, internal errors etc.)
+                Err(other) => {
+                    warn!(
+                        self.logger,
+                        "Exporter: Updating permissioned price accounts failed unexpectedly, using cached value";
+                        "cached_value" => format!("{:?}", self.perm_price_accounts),
+                        "error" => other.to_string(),
+                    );
+                    break;
+                }
+            }
+        }
+
+        // Filter out price accounts we're not permissioned to update
+        let permissioned_updates = fresh_updates
+            .into_iter()
+            .filter(|(id, _data)| {
+                let key_from_id = Pubkey::new(id.clone().to_bytes().as_slice());
+                if self.perm_price_accounts.contains(&key_from_id) {
+                    true
+                } else {
+                    warn!(self.logger, "Exporter: Attempted to publish a price without permission, skipping";
```

Comment:
I don't think that a WARN log be good. Mostly because some publishers publish for both mainnet and pythnet so this is expected to happen. Can we make it an INFO log level?

Reply:
I think you're right, thanks